task/common/values.go

```diff
@@ -41,7 +41,15 @@ type Event struct {
 	Code        string
 	Description []string
 }
-type Storage struct{}
+type RemoteStorage struct {
+	// Container stores the id of the container to be used.
+	Container string
+	// Path stores the subdirectory inside the container.
+	Path string
+	// Config stores provider-specific configuration keys for accessing the pre-allocated
+	// storage container.
+	Config map[string]string
```

Comment:
I wonder if we should take a string and do the parsing internally when needed, so the API looks a bit simpler.

Reply:
no.

Reply:
I'd prefer to be as consistent as possible with the public API (i.e. if the user-facing configuration takes a type, do the same in the library)

Rather subjective, three-character negative replies are in the realms of possibility. 😄

Reply:
I'm not sure I follow

Reply:
I mean that if e.g. the command-line tool takes a string in the form `bucket/path` it might make sense to mirror that structure internally.

Reply:
For instance, have a single struct field of type `string` to hold the `bucket/path` value and then have some struct methods to retrieve the bucket (“container”) and path components. 🤔

Reply:
Both approaches seem valid to me.

Reply:
Ditto :ish: